crates/rspack_binding_values/src/compilation/mod.rs

```diff
@@ -574,6 +558,76 @@ impl JsCompilation {
   pub fn entries(&'static mut self) -> JsEntries {
     JsEntries::new(self.0)
   }
+
+  #[napi]
+  pub fn dispose(&self, env: Env) {
+    // TODO: use napi_add_finalizer if N-API version >= 5
+    let compilation_id = self.0.id();
+
+    COMPILATION_INSTANCE_REFS.with(|refs| {
+      let mut refs = refs.borrow_mut();
+      let r = refs.remove(&compilation_id);
+      if let Some(mut r) = r {
+        let _ = r.unref(env.raw());
+      }
+    });
+
+    MODULE_INSTANCE_REFS.with(|refs| {
+      let mut refs_by_compilation_id = refs.borrow_mut();
+      let refs = refs_by_compilation_id.remove(&compilation_id);
+      if let Some(mut refs) = refs {
+        for (_, mut r) in refs.drain() {
+          let _ = r.unref(env.raw());
+        }
+      }
+    });
+  }
+}
+
+thread_local! {
+  pub static COMPILATION_INSTANCE_REFS: RefCell<HashMap<CompilationId, Ref>> = Default::default();
+}
+
+// The difference between JsCompilationWrapper and JsCompilation is:
+// JsCompilationWrapper maintains a cache to ensure that the corresponding instance of the same Compilation is unique on the JS side.
+//
+// This means that when transferring a JsCompilation from Rust to JS, you must use JsCompilationWrapper instead.
+pub struct JsCompilationWrapper(pub(crate) &'static mut rspack_core::Compilation);
+
+impl JsCompilationWrapper {
```

Comment:
Would you shed more light on this design? What's the benefit of this?

Reply:
Avoid repeatedly create JS Class instances of Compilation.

Reply:
Have you benchmarked it? Do you have the data?

Reply:
`arco-pro_production-mode + exec` case has been `-4.74 %`

Reply:
This pattern is used in `JsStats` here: https://github.com/web-infra-dev/rspack/pull/7344